driver/event_table.c

```diff
@@ -384,6 +384,9 @@ const struct ppm_event_info g_event_info[PPM_EVENT_MAX] = {
 	/* PPME_SYSCALL_BPF_2_X */{"bpf", EC_OTHER, EF_CREATES_FD, 1, { {"fd", PT_FD, PF_DEC} } },
 	/* PPME_SYSCALL_MLOCK2_E */{"mlock2", EC_MEMORY, EF_NONE, 0},
 	/* PPME_SYSCALL_MLOCK2_X */{"mlock2", EC_MEMORY, EF_NONE, 4, {{"res", PT_ERRNO, PF_DEC}, {"addr", PT_UINT64, PF_HEX}, {"len", PT_UINT64, PF_DEC}, {"flags", PT_UINT32, PF_HEX, mlockall_flags}}},
+	/* PPME_SYSCALL_FSCONFIG_E */{"fsconfig", EC_SYSTEM, EF_USES_FD, 0},
```

Comment:
Does using `EF_USES_FD` create trouble on an enter event with 0 parameters?

This is the piece of code I have on top of my mind: https://github.com/falcosecurity/libs/blob/539964408e4e43e94324eace191563d940e8d09a/userspace/libsinsp/parsers.cpp#L708

In cascade, this also causes:
- Parsing reset failure due to `tinfo->m_lastevent_fd` not being set (see: https://github.com/falcosecurity/libs/blob/539964408e4e43e94324eace191563d940e8d09a/userspace/libsinsp/parsers.cpp#L802)
- Fields such as `evt.fd` extraction failure (see: https://github.com/falcosecurity/libs/blob/539964408e4e43e94324eace191563d940e8d09a/userspace/libsinsp/filterchecks.cpp#L1785)

So two options I would suggest:
- Collecting fd as a parameter of the enter event
- Keep the enter event with 0 params, but create a special if case for the exit event of fsconfig to retrieve the fd from the exit event (similarly to what happens with copy_file_range)

Reply:
You are :100: right i think! Will double check everything, thanks!

Reply:
Thank you @jasondellaluce great catch!! What about removing the `EF_USED_FD` from the enter event like in other cases, see `PPME_SYSCALL_IO_URING_ENTER_X` and `PPME_SYSCALL_TO_URING_REGISTER_X`, in the end, this flag should be event-specific and not syscall specific, so since the enter event doesn't use any `fd` probably is better to avoid the flag... WDYT?

Reply:
Yep it is surely needed!